src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
+            perf = new Perfection();
+        }
+        return perf;
+    }
+
+
+    public static boolean isPerfect(long candidate) {
+        boolean retVal;
+        long[] divisors = GetDivisors(candidate);
+        int sum = 0;
+        for (int d = 0; d < 1000; d++)
+        {
+            sum = sum + divisors[d];
+        }
+        if (sum == candidate)
+            retVal = true;
+        return retVal;
+    }
+
+
+    private static long[] GetDivisors(long candidate) {
+        long[] divisors = new long[];
```

Comment:
Missing length of the array. I could be initialized with _candidate_ divided by 2 as it can't have more divisors than that value. 
`long [] divisors = new long[(int)candidate / 2];`

However, having this array seems to be not necessary as you can have a variable in which you can sum every divisor every time you find one. This way you avoid creating this long array.